### DISCUSSION TITLE
Using the Record plugin as the stream input

### DISCUSSION THREAD

**Discussion by cesxhin** [katspaugh/wavesurfer.js#3299] [Q&A] [answered]
When I was using the beta(example: 7.0.0-beta.10), I could use the record plugin as the stream input. But since I updated (version 7.4.2) I can't do this anymore. I need to generate a wave in real time that brings me audio as if it were a microphone but it's not.

In 7.0.0-beta.10

`
    wavesurfer.value = WaveSurfer.create({
        container: '.waveform',
        waveColor,
        height: 400,
        barHeight: 100
    });
`
   `const record = wavesurfer.value.registerPlugin(Recorder.create());`
   `record.render(stream); (from WebRTCMedia)`

**Comment by katspaugh** [comment]
This was never an intended usage. If I’m not mistaken that method was private. But feel free to make a pull request to enable this again.

**Reply by cesxhin** [reply]
Thank your for answer. What do I need to do to make a pull request?

**Reply by katspaugh** [reply, marked as answer]
Probably make a public method that renders an arbitrary stream.
There's already a renderMicStream method which is private.

**Reply by cesxhin** [reply]
I changed private function renderMicStram to public function renderMicStram and work perfect now. How can I open a pull request with my changes made?

**Reply by katspaugh** [reply]
You can fork the repo, clone it, make a commit, push to your fork, then make a PR via this website.

**Reply by cesxhin** [reply]
Done, thank you for support!
